File: Problems/No.283_Move_Zeroes/move_zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """

        # parameter
        pSwap = 0

        # start computing
        for i in range(len(nums)):
            if (nums[i] != 0):
                nums[pSwap], nums[i] = nums[i], nums[pSwap]
                pSwap += 1

File: Problems/No.283_Move_Zeroes/move_zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        
        // parameters
        int pSwap = 0;

        // start computing
        for (int i=0; i<nums.size(); i++) 
            if (nums[i] != 0) std::swap(nums[pSwap++], nums[i]);

    }
};
